internal/service/advancedcluster/data_source.go

```diff
@@ -0,0 +1,32 @@
+package advancedcluster
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+const resourceName = "advanced_cluster" // TODO: if resource exists this can be deleted
```

Comment:
remove comment?

Reply:
i think it's ok to leave it as it helps to know that it has to be removed if resource exists

Reply:
sorry I misunderstood, I thought you meant to delete the comment not the full line.  It's not usual but happens that we can have a data source but not a resource, in that case this is needed, e.g. control_plane_ip_addresses

Reply:
>I thought you meant to delete the comment

I was referring to delete the comment, yes. My point is since the comment is meant to serve as a guidance during scaffolding. Now that we have the resource and data sources figured out, the comment isn't needed anymore because there's nothing left `TODO`

Reply:
for it's a guide for the tool user, we don't do complex logic yet, so if you have a resource and data source you can see the comment and understand why it's there and delete it,   right now we don't have logic to keep the line only if datasource exists